From b45edd940ba17e0256f94feb767a7455a80e2d9d Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Wed, 13 Jan 2021 08:27:13 +0200
Subject: [PATCH] [SW-20154] make base addresses platform specific

Move addresses such as CFG Base and offsets to
flash etc to platform specific code, which will
enable to add support for Gaudi2.
Along with these, move the key size related
defines as well.

Change-Id: Ia8f2784e5cbf395a9504108466193483414c41a0
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/common/secureboot.h b/common/secureboot.h
index a69bb63..74d3537 100644
--- a/common/secureboot.h
+++ b/common/secureboot.h
@@ -9,7 +9,6 @@
 #define _SECUREBOOT_H_
 
 #define SIG_SIZE			384
-#define KEY_SIZE			392
 #define IMG_LEN_SIZE			4
 #define FW_VERSION_SIZE			4
 #define CRC32_SIZE			4
diff --git a/goya2_arc/soc.h b/goya2_arc/soc.h
index a986555..6ebd1a5 100644
--- a/goya2_arc/soc.h
+++ b/goya2_arc/soc.h
@@ -14,4 +14,23 @@
 #define PREBOOT_ADDR_H	0x7F
 #define PREBOOT_ADDR_L	0xF0000000
 
+#define FW_SIZE					65535
+#define SIG_SIZE				384
+#define KEY_SIZE				392
+#define BLOB_START				4
+
+#define SPI_FLASH_BLOCK_SIZE			1024
+
+#define HL_SRAM_BASE				0x7FF0000000ULL
+#define HL_SP_SRAM_BASE            		0x7FFBFE0000ULL
+#define HL_SDRAM_BASE				0x10000000
+#define HL_FLASH_OFFSET				0xA000000
+#define HL_ICCM_OFFSET				0xB020000
+#define HL_SBROM_OFFSET				0xB040000
+#define HL_FWSRAM_OFFSET			0xB050000
+#define HL_SCRATCHPAD_SRAM_OFFSET		0xBFE0000
+#define HL_PCIE_FWSRAM_OFFSET			0xBFF0000
+
+#define HL_CFG_BASE				0xFC000000
+
 #endif /* _SOC_H_ */
diff --git a/include/arcboot.h b/include/arcboot.h
index faffd00..98ad433 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -14,23 +14,6 @@
 #define BIT(nr)					(1 << (nr))
 #define ARRAY_SIZE(x)				(sizeof(x) / sizeof((x)[0]))
 
-#define FW_SIZE					65535
-#define SIG_SIZE				384
-#define KEY_SIZE				392
-#define BLOB_START				4
-
-#define SPI_FLASH_BLOCK_SIZE			1024
-
-#define HL_SRAM_BASE				0x7FF0000000ULL
-#define HL_SP_SRAM_BASE            		0x7FFBFE0000ULL
-#define HL_SDRAM_BASE				0x10000000
-#define HL_FLASH_OFFSET				0xA000000
-#define HL_ICCM_OFFSET				0xB020000
-#define HL_SBROM_OFFSET				0xB040000
-#define HL_FWSRAM_OFFSET			0xB050000
-#define HL_SCRATCHPAD_SRAM_OFFSET		0xBFE0000
-#define HL_PCIE_FWSRAM_OFFSET			0xBFF0000
-
 #define HL_FLASH_BASE				(HL_SDRAM_BASE + \
 						 HL_FLASH_OFFSET)
 #define HL_SBROM_BASE				(HL_SDRAM_BASE + \
@@ -39,9 +22,6 @@
 						 HL_FWSRAM_OFFSET)
 #define HL_SCRATCHPAD_SRAM_BASE			(HL_SDRAM_BASE + \
 						 HL_SCRATCHPAD_SRAM_OFFSET)
-
-#define HL_CFG_BASE				0xFC000000
-
 #define ARCBOOT_PRIMARY				0
 #define ARCBOOT_SECONDARY			1
 
